homeassistant/components/google/calendar.py

```diff
@@ -341,30 +337,50 @@ def _event_filter(self, event: Event) -> bool:
     async def async_added_to_hass(self) -> None:
         """When entity is added to hass."""
         await super().async_added_to_hass()
+
         # We do not ask for an update with async_add_entities()
-        # because it will update disabled entities
-        await self.coordinator.async_request_refresh()
-        self._apply_coordinator_update()
+        # because it will update disabled entities. This is started as a
+        # task to let if sync in the background without blocking startup
+        async def refresh() -> None:
+            await self.coordinator.async_request_refresh()
+            self._apply_coordinator_update()
+
+        self.hass.async_create_task(refresh())
```

Comment:
During the Home Assistant setup phase tasks created with `hass.async_create_task` are tracked and setup will wait for them for a timeout period before continuing. If we don't want to wait at all we should use `asyncio.create_task`.

Reply:
Thanks, fixed. I was going back and forth about whether this really should block or not and still lean towards having it not block.  I think there could be more i do here to remove hard cloud dependencies (e.g. sync calendar list), more testing, etc, so may do more in follow ups.